tests/ui_tests/test_ui_add_component_window.py

```diff
@@ -827,6 +1134,349 @@ def test_UI_GIVEN_array_field_selected_and_edit_button_pressed_THEN_edit_dialog_
     assert field.table_view.isEnabled()
 
 
+def test_UI_GIVEN_user_selects_pixel_grid_THEN_pixel_grid_is_set_to_true_in_ok_validator(
```

Comment:
There's a lot of repeated code in the following 10 tests. Don't be afraid to create functions to do the common bits